packages/algob/test/stubs/algo-operator.ts

```diff
@@ -22,25 +23,44 @@ export class AlgoOperatorDryRunImpl implements AlgoOperator {
     throw new Error("Not implemented");
   }
 
-  getAssetByID (assetIndex: number | bigint): Promise<import("algosdk").AssetInfo> {
-    throw new Error("Not implemented");
+  getAssetByID (assetIndex: number | bigint): Promise<AssetInfo> {
+    if (assetIndex === 1) {
+      const res: AssetInfo = {
+        index: 1,
+        params: {
+          creator: "addr-1",
+          total: 1000,
+          decimals: 8,
+          'default-frozen': false,
+          'unit-name': "TKN",
+          name: "ASA-1",
+          url: "link",
+          'metadata-hash': "12312442142141241244444411111133",
+          manager: bobAcc.addr,
+          reserve: undefined,
+          freeze: bobAcc.addr,
+          clawback: undefined
+        }
+      };
+      return new Promise((resolve, reject) => {
+        resolve(res);
+      });
+    } else {
+      throw new Error("Not implemented");
```

Comment:
If we use `promise.resolve(res)` here, shouldn't we use `promise.reject("Not implemented")` here ? I don't exactly know the reject mechanism, does it not throw error, & simply return?

Reply:
it will throw error, because function is not going to resolve, Before resolving it throws an error.

Reply:
Ok so i thought this could be:
```js
return new Promise((resolve, reject) => {   
 assetIndex === 1 ? resolve(res); : reject("Not implemented");      
});

Reply:
will check this one.